### ISSUE TITLE
Tools > Addons: to open in Commando > Package top level scope

### ISSUE THREAD

**Issue by cgchoffman** [Komodo/KomodoEdit#897]
Just dealt with a user support request where they said a package wasn't available in Komodo why they tried to use Tools > Addons and Commando opened.

Though it does say "Installed Packages"  if you're not opening the dialog for that reason it's a little confusing.  

I think it would be better to open it in the Package top level scope so you can just start typing and determine the scope you want as you go.  Either the package is installed and you manage it or it's not and you install it.  With the current implementation, you have to jump up a scope if you want something other than uninstalling a plugin.

I think installing addons is the more common process any way.

**Comment by Naatan** [comment]
Don't forget to target & label.

**Comment by garretwilson** [comment]
+1 (I couldn't find out how to vote on this.)

**Comment by Defman21** [comment]
You don't have to vote for bugs. There is no reputation system based on users votes.

**Comment by Naatan** [comment]
It's fine @Defman21, it tells us what bugs need prioritizing based on user feedback.

**Comment by Defman21** [comment]
GitHub could implement that.. :)

**Comment by Naatan** [comment]
Don't hold your breath.

**Comment by garretwilson** [comment]
https://www.change.org/p/github-inc-add-voting-functionality-for-github-issues

**Comment by Naatan** [comment]
Unfortunately nothing says "never going to happen" like having a change.org petition for it :\

**Comment by garretwilson** [comment]
Haha. :thumbsup: (Sorry, couldn't resist...)

**Comment by garretwilson** [comment]
To add some feedback to help you with improving the add-ons dialog, I'm really having trouble figuring out how this is supposed to work.

After discussion on the forum I found out that I have to hit the secret/magic backspace key to get a list of things to do. But there are other problems.

* If I click the dropdown list and select "Everything", I get a magnifying class and I can search/filter everything. OK, that's reasonable.
* But if I close the dialog and then go back to **Tools > Add-ons** after this, I get a magnifying glass, the words "Installed Packages >", and a back-and-forth animated background color. First of all, **Tools > Add-ons** should always bring me back to the same place---it shouldn't remember the state of the dialog. Secondly, it says to hit the right arrow key to expand my selection, but hitting right-arrow does nothing.
* Strangely if I select the dropdown list and go to "Packages" at this point, the dialog disappears altogether!
* If I finally find my installed Add-ons (e.g. Beautify js) and double-click one of them, it seems to re-install my add-on! I was hoping to get the configuration. At the very least it should ask me if I want to re-install. And maybe I wanted to uninstall---or get the configuration. I'm not sure what it should be doing at this point, but if it's going to reinstall something I already have installed, I would expect it to ask me first.

This Add-ons dialog is a very strange beast---very buggy and confusing.

And I still haven't figure out how to pull up the configuration for the Beautify js add-on.

**Comment by cgchoffman** [comment]
@garretwilson  That's great feedback but can you file a different enhancement request? This is specifically for the **Tools > Addon** menu item.

**Comment by garretwilson** [comment]
Yes, @cgchoffman , I would be happy to file a different enhancement request, but you response leaves me puzzled. All my feedback above was specifically about the **Tools-Addons** menu and related bugs/inconsistencies.

**Comment by cgchoffman** [comment]
@garretwilson your feedback is on **go to anything** UX.  Go to Anything (AKA Commando) is more than just addons.  This enhancement request is specifically about what happens when you click **Tools > Addons**.

**Comment by garretwilson** [comment]
> our feedback is on go to anything UX.

As a user I don't know what a "go to anything UX" is. All I know is that I used the **Tools > Add-ons** menu to get to all the stuff I mentioned.

Maybe one of the bugs is that **Tools > Add-ons** brings up all this other stuff, and then makes you get lost in the other stuff trying to find how to add new add-ons or configure add-ons.

Plus the bug I mentioned about using **Tools > Add-ons** and it bringing up the magnifying glass is a bug, and it is directly related to **Tools > Add-ons**. Please re-read my bulleted points above carefully.

> This enhancement request is specifically about what happens when you click Tools > Addons.

Everything I mentioned is what happened after I clicked on **Tools > Add-ons**.

**Comment by cgchoffman** [comment]
I have read you bullet points and I agree they are valid, I just need you to report them in different issues so we can track them properly

> But if I close the dialog and then go back to Tools > Add-ons after this, I get a magnifying glass, the words "Installed Packages >"

This is a bug I agree.  It is a different bug though so again, please file another bug.

> Strangely if I select the dropdown list and go to "Packages" at this point, the dialog disappears altogether!

Again, different bug.  Please file a different report.  

> If I finally find my installed Add-ons (e.g. Beautify js) and double-click one of them, it seems to re-install my add-on! I was hoping to get the configuration.

This is UX (user experience, apologies if I'm using an acronym you're not familiar with) that I'm talking about.  This needs to be reported and discussed in a separate issue.

Please don't take this as negative.  This is awesome.  But we need to keep this organized.  Your activity is hugely appreciated.

**Comment by garretwilson** [comment]
> This is UX (user experience, apologies if I'm using an acronym you're not familiar with)

I got that. I was merely pointing out that which exact panel in the source code this referred to was more of a developer-oriented concern; as a user all that concerns me is that I go to **Tools > Add-ons**... and get lost. :)

> I just need you to report them in different issues so we can track them properly

OK, no worries---that I completely understand!

> Please don't take this as negative. This is awesome. But we need to keep this organized. Your activity is hugely appreciated.

I sincerely appreciate that you're taking all my complaining as constructive, because that's how I mean them. I hope I can help! I'll go file those bugs now.

**Comment by cgchoffman** [comment]
If I could "like" your comment I would....+1.  Thanks @garretwilson.

**Comment by Defman21** [comment]
There's a thumb up emoji, so you actually can give this sir a :+1:

**Comment by Naatan** [comment]
You and your logic.

**Comment by garretwilson** [comment]
Haha. So we're back on the "like" thread again. ;) :+1:

**Comment by garretwilson** [comment]
OK, so I've filed Issue #960, Issue #961, and Issue #962, all related to this issue.

Frankly, though, I think the whole add-ons UI is a bit confusing. Why don't you check out the add-ons pages of Firefox and Thunderbird? Much more intuitive.

Even Atom Editor does this much better than Komodo, although it's not as good as Firefox and Thunderbird in this regard.

**Comment by Naatan** [comment]
The new package manager interface was introduced only a few months ago, it's only at the very start of its evolution.
